frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -251,6 +299,17 @@ const useServiceDeployment = () => {
     overrideSelectedServiceStatus(MiddlewareDeploymentStatus.DEPLOYING);
 
     try {
+      const isHealthy = await performHealthCheck(serviceId);
+
+      // If not healthy, reset the status and do not proceed
+      if (!isHealthy) {
+        overrideSelectedServiceStatus(selectedService?.deploymentStatus);
+        return;
+      }
+
+      console.log('Starting service... (SHOULD NOT HAPPEN)');
+      return;
+
```

Comment:
I don't think we should do that, MW will still think the agent is running and the agent will be running, why adding an unclear state on FE?